lib/engines/content_block_manager/app/models/content_block_manager/content_block/document/soft_deletable.rb

```diff
@@ -2,6 +2,10 @@ module ContentBlockManager
   module ContentBlock::Document::SoftDeletable
     extend ActiveSupport::Concern
 
+    included do
+      default_scope { where(deleted_at: nil) }
```

Comment:
really like the approach of adding `deleted_at` in general, but wondering if there should be some change to the `state` of the document's editions? e.g. they are now in a `deleted` state. Or do you think that's a bit too complicated at this point?

I'm also wondering if we keep all the scoping logic in once place - we have the `live` scope here that the `DocumentFilter` uses https://github.com/alphagov/whitehall/blob/eb94bca2a74591c37fb8f3f29241b36ef83f2cc9/lib/engines/content_block_manager/app/models/content_block_manager/content_block/document.rb#L29

Reply:
I liked keeping the default scope in the `SoftDeletebale` concern as it relates to things being soft deleted and communicates that better I think. Without the default scope in there, you could remove the default scope from the model and it would still look like things were being soft deleted, but all it would be doing in effect is adding a timestamp to a column.

The state stuff I did consider, but like you say, I think that might be a bit too complex for now.